### ISSUE TITLE
Train Test Split documentation

### ISSUE THREAD

**Issue by ardyadipta** [google/meridian#680]
On the train test split [documentation](https://developers.google.com/meridian/docs/user-guide/configure-model#use-training-test-data-split), It does not consider the time series nature of the data. It randomly selects time indices across the entire time range, which can result in the test set containing time points before or interspersed with the training set, violating the chronological order required for time series modeling.

As far as I know, in MMM typically want to predict future outcomes based on past data, so the standard is to use earlier time points as training set, and use the later time points as the test set to avoid data leakage and more accurately simulates real world forecasting.

Or is there any assumption in Meridian that I should know? 
Thanks!

**Comment by gousesb** [comment]
Hello @ardyadipta ,

Thank you for contacting us!

Meridian is a highly customizable Bayesian causal inference modeling framework that offers insights and visualizations for marketing decisions. The primary goal of Meridian is to accurately estimate the causal effects of marketing, like the casual relationships between the treatments and the target KPI (say ROI). While prediction is a byproduct, it's not the central aim. It allows users to estimate the causal impact of their media execution and non-media spends.

Meridian is not suitable for forecasting KPIs exhibiting strong trends and seasonality as it focuses on estimating causal media impact using a knot-based method to model trend and seasonality. This method requires data points near the knots for accurate estimation. Therefore, holding out large, contiguous periods of data leads to lack of data near knots during held-out periods resulting in unreliable estimates as the knots’ prior distribution heavily influences the posterior distribution. Hence, using Meridian for forecasting is not recommended.

For more information regarding holdout_id, please refer to our documentation on [Holdout observations](https://developers.google.com/meridian/docs/basics/holdout-observations).

Feel free to reach out for any further queries or feedback regarding Meridian.

Thank you

Google Meridian Support Team

**Comment by ardyadipta** [comment]
Thanks for the reply!

However, if the marketing team in the company I work for would like to use
Meridian for "What-If" scenarios in the future, what would you suggest?
For example, for the upcoming summer holiday season, which channel should
they invest more given the trained MMM model?

In the past project while using PyMC I could create an interface using
streamlit, so that the marketing team could tune the spend for each channel
and then I could predict the ROI given the beta of each channel is already
well estimated. Additionally we also create an input for seasonality such
as whether there is a promo or holiday season.

Thank you!
Dipta

**Comment by AdimDrewnik** [comment]
Response curves in both Meridian and PyMC don't interact with seasonality, so it doesn't matter which part of the year is upcoming as you will get the same prediction from channels. 
Meridian has functions to give you response curves so you can extract them and give them to marketing team to play with.

**Comment by gousesb** [comment]
Hello Dipta,

Thank you for contacting us!

To analyze the results and determine the suggested budget allocations in what-if scenarios, we currently recommend two methods depending on the data availability.

1. If you don’t have data available for a future time-period containing the expected media execution, then we recommend selecting a period from the past that matches the expected flighting patterns of the target period and use that (using the `selected_times` argument of the Analysis module methods) to analyze the results and optimize budget accordingly. For example, if you wish to determine budget optimization for the upcoming summer holiday season, then you may choose the previous summer as the period with similar flighting pattern. You may check the previous discussion [#267](https://github.com/google/meridian/issues/267#issuecomment-2332702599) which describes this process in greater detail.

2. Meridian  allows using the `new_data` argument in Analyzer class methods that allows you to analyze results predicted by the model for unseen data. For example, the BudgetOptimizer function can determine budget allocations for the what-if scenarios using the `new_data` argument in the [`optimizer()` function](https://developers.google.com/meridian/reference/api/meridian/analysis/optimizer/BudgetOptimizer#optimize).

      If new_data is provided, the optimization is run on the versions of the tensors in new_data and the original versions of all the remaining tensors (i.e., you will only need to provide media data in the `new_data` and controls, etc. are chosen from the original data). If any of the tensors in new_data is provided with a different number of time periods than in InputData, then all tensors must be provided with the same number of time periods and the time tensor must be provided. We can customize this also by providing media, reach, frequency, media_spend, rf_spend, revenue_per_kpi, and time.

Feel free to reach out for any further queries or feedback regarding Meridian.

Thank you

Google Meridian Support Team
